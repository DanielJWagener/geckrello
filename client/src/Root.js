// Simply creates a Redux store and passes it in to a provider tag.
// This logic was originally in index.js, but we've movied it here to
// make our testing process easier.
import React from "react";
import { Provider } from "react-redux";
import { createStore, applyMiddleware, compose } from "redux";
import reduxThunk from "redux-thunk";
import reducers from "./reducers";

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const store = createStore(
  reducers,
  {},
  composeEnhancers(applyMiddleware(reduxThunk))
);

export default ({ children }) => {
  return <Provider store={store}>{children}</Provider>;
};
