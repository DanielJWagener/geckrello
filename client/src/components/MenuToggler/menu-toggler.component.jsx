import React from "react";

import "./menu-toggler.styles.scss";

const MenuToggler = props => {
  // Changes input styling on mouse over
  const onMouseOverHandler = event => {
    event.target.className = "toggler toggler--hover";
  };

  // Changes input styling on mouse out
  const onMouseOutHandler = event => {
    event.target.className = "toggler";
  };

  return (
    <div>
      <h1
        onClick={props.toggleSidebar}
        onMouseOver={onMouseOverHandler}
        onMouseOut={onMouseOutHandler}
        className="toggler"
      >
        {props.sidebarHidden ? "Show Menu" : "Hide Menu"}
      </h1>
    </div>
  );
};

export default MenuToggler;
