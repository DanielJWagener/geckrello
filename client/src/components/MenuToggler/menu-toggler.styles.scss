@import "../../sass/abstracts/_variables.scss";

.toggler {
  color: white;
  cursor: pointer;
  border-radius: 5px;

  padding: 3px 0.5rem;
  margin-right: 1rem;
  border: 2px solid var(--color-primary-light);
  text-align: center;

  h1 {
    font-weight: 100;
  }

  &--hover {
    background-color: var(--color-primary-hoverLight);
    border: 2px solid var(--color-primary-hoverLight);
  }

  @media only screen and (max-width: 23.75em) {
    margin-right: 0.5rem;
  }
}
