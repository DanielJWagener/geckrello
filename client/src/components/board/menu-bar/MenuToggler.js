import React from "react";

const MenuToggler = props => {
  // Changes input styling on mouse over
  const onMouseOverHandler = event => {
    event.target.className = "menu-bar__toggler menu-bar__toggler--hover";
  };

  // Changes input styling on mouse out
  const onMouseOutHandler = event => {
    event.target.className = "menu-bar__toggler";
  };

  return (
    <div>
      <h1
        onClick={props.toggleSidebar}
        onMouseOver={onMouseOverHandler}
        onMouseOut={onMouseOutHandler}
        className="menu-bar__toggler"
      >
        {props.sidebarHidden ? "Show Menu" : "Hide Menu"}
      </h1>
    </div>
  );
  /*
  if (props.sidebarHidden) {
    return (
      <div className="menu menu-bar__toggler">
        <h1 onClick={props.toggleSidebar} onMouseOver={onMouseOverHandler} onMouseOut={onMouseOutHandler}>{props.sidebarHidden ? "Show Menu" : "Hide Menu"}</h1>
      </div>
    );
  } else {
    return (
      <div className="menu menu-bar__toggler">
        <h1 onClick={props.toggleSidebar}>Hide Menu</h1>
      </div>
    );
  }
  */
};

export default MenuToggler;
