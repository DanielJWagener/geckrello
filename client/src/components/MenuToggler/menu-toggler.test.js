import { shallow } from "enzyme";
import React from "react";
import MenuToggler from "./menu-toggler.component";
import Root from "../../Root";

it("renders MenuToggler component", () => {
  expect(shallow(<MenuToggler />).debug()).toMatchSnapshot();
});
