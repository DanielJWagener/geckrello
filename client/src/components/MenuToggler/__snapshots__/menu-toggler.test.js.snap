// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MenuToggler component 1`] = `ShallowWrapper {}`;
